File: ChipWhisperer/poly_masking_setup/poly_masking_parameters_d1e1k1.py
# SPDX-License-Identifier: MIT
# Copyright 2025-2026 Max Planck Institute for Security and Privacy (MPI-SP), University of Luebeck
secrets_supports = [0]
shares_supports = [188, 189, 1]
# Permutation map for shares squaring in GF(2^8)
shares_permutation_map = [1, 0, 2]
#Permutation map for secrets squaring in GF(2^8)
secrets_permutation_map = [0]
# Vandermonde matrix in GF(2^8)
V = [
    [1, 188, 189],
    [1, 189, 188],
    [1, 1, 1]
]
# Inverse Vandermonde matrix in GF(2^8)
V_inv = [
    [1, 1, 1],
    [189, 188, 1],
    [188, 189, 1]
]
M_enc = [
    [189, 188],
    [188, 189]
]
M_dec = [
    [1, 1, 1]
]
A_tilde = [
    [
        [1],
        [188],
        [189]
    ]
]

File: qemu_implementation/build_poly_masking_parameters/poly_masking_parameters_d1e1k1.c
// SPDX-License-Identifier: MIT
// Copyright 2025-2026 Max Planck Institute for Security and Privacy (MPI-SP), University of Luebeck
#include "../src/poly_masking_parameters.h"
#include <stdint.h>
__attribute__((section(".data"))) uint8_t secrets_supports[1] = {0};
__attribute__((section(".data"))) uint8_t shares_supports[3] = {188, 189, 1};
// Permutation map for shares squaring in GF(2^8)
__attribute__((section(".data"))) uint8_t shares_permutation_map[3] = {1, 0, 2};
// Permutation map for secrets squaring in GF(2^8)
__attribute__((section(".data"))) uint8_t secrets_permutation_map[1] = {0};
// Vandermonde matrix in GF(2^8)
__attribute__((section(".data"))) uint8_t V[3][3] = {
    {1, 188, 189},
    {1, 189, 188},
    {1, 1, 1}
};
// Inverse Vandermonde matrix in GF(2^8)
__attribute__((section(".data"))) uint8_t V_inv[3][3] = {
    {1, 1, 1},
    {189, 188, 1},
    {188, 189, 1}
};
__attribute__((section(".data"))) uint8_t M_enc[2][2] = {
    {189, 188},
    {188, 189}
};
__attribute__((section(".data"))) uint8_t M_dec[1][3] = {
    {1, 1, 1}
};
__attribute__((section(".data"))) uint8_t A_tilde[1][3][1] = {
    {
        {1},
        {188},
        {189}
    }
};
__attribute__((section(".data"))) uint8_t lambda_hat[3][3] = {
    {1, 1, 1},
    {1, 1, 1},
    {1, 1, 1}
};
